Tests/Unit/Common/XmlDocumentTest.php

```diff
@@ -11,27 +11,50 @@
 final class XmlDocumentTest extends UnitTestCase
 {
     private XmlDocument $subject;
+    private $xmlString;
 
     protected function setUp(): void
     {
         parent::setUp();
 
-        $this->subject = new XmlDocument('');
+
+        $this->xmlString = file_get_contents('Tests/Testfiles/meitest2.xml');
+        $this->subject = XmlDocument::from($this->xmlString);
+
     }
 
     /**
      * @test
      */
     public function returnsArray(): void
     {
-        self::assertSame([], $this->subject->toArray());
+        self::assertIsArray($this->subject->toArray());
     }
 
     /**
      * @test
      */
     public function returnsJson(): void
     {
-        self::assertSame('', $this->subject->toJson());
+        self::assertIsArray($this->subject->toJson());
+    }
+
+    /**
+     * @test
+     */
+
+    public function xmlStringNotEmpty(): void
+    {
+        self::assertNotSame('', $this->xmlString);
     }
+
+    /**
+     * @test
+     */
+
+    public function xmlStringEqualFile() {
+        self::assertXmlStringEqualsXmlFile('Tests/Testfiles/meitest2.xml',$this->xmlString);
+    }
+
```

Comment:
provide more tests (probably not making use of files) with minimal examples:
- convert attribute
- convert text value
- convert mixed content
- all config options